autogpt_platform/autogpt_libs/autogpt_libs/supabase_integration_credentials_store/types.py

```diff
@@ -8,6 +8,7 @@ class _BaseCredentials(BaseModel):
     id: str = Field(default_factory=lambda: str(uuid4()))
     provider: str
     title: Optional[str]
+    hidden: Optional[bool] = False
```

Comment:
Putting this here will also persist it in the DB, which I think should be avoided if possible.

IMO there should be minimal involvement of the credentials system when a user wants to use credits to run a block. Right now we're injecting credentials into the credential system in a way it definitely wasn't designed for and that doesn't make sense to me as anything but a quick fix (which is what it was). This PR preferably shouldn't further entrench this solution by expanding on it.

I'd extend the `CredentialsInput` with a `credits_available: bool | list[CP]` parameter. Then the frontend can render an option "Use credits" in the credentials picker, which (when selected) sets `credentials = null`, or some magic value like `credentials = {id: 0, provider: "agpt-credits"}`.

Reply:
Another option would be to put the `credentials` and `model` input together in a sub-input, annotate the `model` as a discriminator, and implement some JSON Schema based discrimination logic in the frontend.

Reply:
`hidden` is removed. As for the `CredentialsInput` extension and merging `credentials` and `model` (I also thought about this and it is a good solution imo) I think that'a outside the PR scope here.